lib/incoming_form.js

```diff
@@ -82,7 +82,19 @@ IncomingForm.prototype.parse = function(req, cb) {
     var fields = {}, files = {};
     this
       .on('field', function(name, value) {
-        fields[name] = value;
+        if (name.slice(-2) === '[]') {
+          var realName = name.slice(0, name.length - 2);
+          if (realName in fields) {
```

Comment:
`fields` is an `Object`, so `realName`s on `Object.prototype` will produce strange results. `Object.prototype.hasOwnProperty.call(fields, realName)` is the more consistent way to do this.

Reply:
Yea, exactly, was thinking about that too.